include/sundials/sundials_iterative.h

```diff
@@ -91,8 +115,8 @@ enum { SUN_MODIFIED_GS = 1, SUN_CLASSICAL_GS = 2 };
  */
 
 /* DEPRECATED ATimesFn: use SUNATimesFn */
-typedef int (*ATimesFn)(void *A_data, N_Vector v, N_Vector z);
-typedef int (*SUNATimesFn)(void *A_data, N_Vector v, N_Vector z);
+typedef SUNLsStatus (*ATimesFn)(void* A_data, N_Vector v, N_Vector z);
```

Comment:
I'm not sure that I understand why `SUNATimesFn`, `SUNPSetupFn`, `SUNPsolveFn`, ... return a `SUNLsStatus` instead of a `SUNErrCode`, since these are internal SUNDIALS utility routines and not `SUNLinearSolver` functions.

Reply:
These are not internal, these are typedefs of prototypes for functions that a user could provide and our API asks them to return codes which may recoverable... thus they are ` SUNLsStatus` not a `SUNErrCode` (which is only unrecoverable).